pandas/core/frame.py

```diff
@@ -993,7 +993,7 @@ def to_dict(self, orient='dict', into=dict):
                           for k, v in compat.iteritems(self))
         elif orient.lower().startswith('r'):
             return [into_c((k, _maybe_box_datetimelike(v))
-                           for k, v in zip(self.columns, row))
+                           for k, v in zip(self.columns, np.atleast_1d(row)))
                     for row in self.values]
```

Comment:
shouldn't the ``np.atleast_2d`` be around ``self.values``?

Reply:
No, it fails then. In that case only one of the results will be returned, effectively flattening the array.